src/server/db/models/testcenter.js

```diff
@@ -29,10 +29,30 @@ module.exports = (sequelize, DataTypes) => {
       closed: DataTypes.BOOLEAN,
       date_closed: DataTypes.DATE,
     },
-    {}
+    {
+      hooks: {
+        beforeCreate: (testCenter) => {
+          return updateGeolocation(testCenter)
+        },
+        beforeUpdate: (testCenter) => {
+          return updateGeolocation(testCenter)
+        },
+      },
+    }
   )
   TestCenter.associate = function (models) {
     // associations can be defined here
   }
+
+  const updateGeolocation = async (testCenter) => {
+    if (testCenter.changed('latitude') || testCenter.changed('longitutde')) {
+      if (testCenter.longitude === null || testCenter.latitude === null) return
```

Comment:
is this an empty return statement, or is this meant to return the object on the next line?

Reply:
maybe add brackets after the if statement, for clarity?

Reply:
Will update. If latitude or longitude are null, we can't set the geometry value. Will bracket.